### PR TITLE
Set nulls correctly for all type of arrays/vectors

### PR DESCRIPTION
**Pull request by y-f-u**
This sets the correct nulls for strings, struct, list, array types when copying data from arrow array to duckdb vectors.

When inserting nulls to columns with these types, the null will be correctly set instead of the default value like: `""`, `{k: ""}`, etc.

### CODE REVIEW

File: crates/duckdb/src/vtab/vector.rs [hunk 1]
```diff
@@ -279,4 +295,13 @@ impl StructVector {
         let logical_type = self.logical_type();
         unsafe { duckdb_struct_type_child_count(logical_type.ptr) as usize }
     }
+
```
Comment (Swoorup) [inline crates/duckdb/src/vtab/vector.rs:298]: Some of these makes no sense. The struct vector doesn't store the columns directly. Think that goes for all Arrays, Union,

Comment (y-f-u) [reply crates/duckdb/src/vtab/vector.rs:298]: Thanks for the comment. This change is about to set the validity on the vector, so duckdb knows when to set null correctly.

See added tests.

**Comment by Mytherin** [comment]
Thanks for the PR! LGTM - could you just merge with main so the CI can rerun?

**Comment by Mytherin** [comment]
Thanks - looks like the CI now fails on the newly added test. Could you have a look when you have time?

**Comment by y-f-u** [comment]
yeah, will do. Sorry got carried away by something else, looks like an easy fix.

**Comment by y-f-u** [comment]
that's still something wrong with invalid cfg feature for time. Which is not introduced by this branch, I took a shot on fixing that but not really sure how the `time` features come from and how it got used.

**Comment by Mytherin** [comment]
Thanks for looking into it - could you merge with main again? The CI should be fixed again in https://github.com/duckdb/duckdb-rs/pull/375

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**